src/platform/input/game-pads.js

```diff
@@ -68,62 +109,619 @@ const MAPS = {
+     */
+    get axes() {
+        return this.map.axes ? this.map.axes.map(a => this.pad.axes[MAPS_INDEXES.axes[a]] || 0) : [];
+    }
+
+    /**
+     * @type {GamePadButton[]} - The buttons present on the GamePad. Some buttons may be null.
+     */
+    get buttons() {
+        return this.map.buttons ? this.map.buttons.map(b => this._getButton(b)) : [];
+    }
+
+    /**
+     * Make the gamepad vibrate.
+     *
+     * @param {number} intensity - Intensity for the vibration in the range 0 to 1.
+     * @param {number} duration - Duration for the vibration in milliseconds.
+     * @param {object} [options] - Options for special vibration pattern.
+     * @param {string} [options.type] - Type of pattern. Available types are "dual-rumble" and "vibration". Defaults to "vibration".
+     * @param {number} [options.startDelay] - Delay before the pattern starts, in milliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators in the range 0 to 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators in the range 0 to 1. Defaults to intensity.
+     * @returns {Promise<boolean>} Return a Promise resulting in true if the pulse was successfully completed.
+     */
+    pulse(intensity, duration, options) {
+        const actuator = this.pad.vibrationActuator;
+
+        if (actuator) {
+            if (actuator.playEffect) {
+                if (options) {
+                    return actuator.playEffect(options.type === 'dual-rumble' ? 'dual-rumble' : 'vibration', {
+                        'duration': duration,
+                        'startDelay': options.startDelay || 0,
+                        'strongMagnitude': typeof options.strongMagnitude === 'number' ? options.strongMagnitude : intensity,
+                        'weakMagnitude': typeof options.weakMagnitude === 'number' ? options.weakMagnitude : intensity
+                    });
+                }
+
+                return actuator.playEffect('vibration', {
+                    'duration': duration,
+                    'startDelay': 0,
+                    'strongMagnitude': intensity,
+                    'weakMagnitude': intensity
+                });
+            } else if (actuator.pulse) {
+                if (options && options.startDelay) {
+                    // Custom delay
+                    return new Promise(function (resolve, reject) {
+                        setTimeout(function () {
+                            actuator.pulse(intensity, duration).then(resolve).catch(reject);
+                        }, options.startDelay);
+                    });
+                }
+
+                return actuator.pulse(intensity, duration);
+            }
+        }
+
+        return Promise.resolve(false);
+    }
+
+    /**
+     * Return true if the gamepad is connected.
+     *
+     * @returns {boolean} Return true if the gamepad is connected.
+     */
+    isConnected() {
```

Comment:
Should be a getter too to keep it coherent

Reply:
It's fixed!